app/src/main/res/layout/activity_nfc.xml

```diff
@@ -0,0 +1,136 @@
+<?xml version="1.0" encoding="utf-8"?>
+<LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:id="@+id/root"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    android:orientation="vertical"
+    tools:context=".nfc.NfcActivity">
+
+    <androidx.appcompat.widget.Toolbar
+        android:id="@+id/toolbar"
+        android:layout_width="match_parent"
+        android:layout_height="?attr/actionBarSize"
+        android:elevation="4dp"
+        android:theme="@style/ThemeOverlay.HomeAssistant.ActionBar" />
+
+    <ProgressBar
+        android:id="@+id/nfc_loading"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:layout_marginTop="100dp" />
+
+
+    <LinearLayout
+        android:id="@+id/mode_switch"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:weightSum="2">
+
+        <androidx.appcompat.widget.AppCompatButton
+            android:id="@+id/add_nfc_tag_id"
+            style="@style/Widget.HomeAssistant.Button.Colored"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:layout_weight="1"
+            android:text="@string/nfc_button_add_tag_id" />
+
+        <androidx.appcompat.widget.AppCompatButton
+            android:id="@+id/add_universal_link"
+            style="@style/Widget.HomeAssistant.Button.Colored"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:layout_weight="1"
+            android:text="@string/nfc_button_add_universal_app_link" />
+    </LinearLayout>
+
+
+    <LinearLayout
+        android:id="@+id/nfc_tag_id_setup"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:gravity="center">
+
+        <androidx.appcompat.widget.AppCompatButton
```

Comment:
Rather than another button why not just give a spinner with instructions to tap to the NFC tag to write too?

Reply:
Great idea, changed it in the latest commit :)